mediathread/assetmgr/migrations/0010_auto_20180221_1700.py

```diff
@@ -0,0 +1,50 @@
+# -*- coding: utf-8 -*-
+# Generated by Django 1.11.10 on 2018-02-21 17:00
+from __future__ import unicode_literals
+
+from os.path import basename, splitext
+from django.db import migrations
+
+
+def fix_permalinks(sources):
+    for source in sources:
+        filename = basename(source.url)
+        if splitext(filename)[1] == '':
+            # This is a broken permalink. Fix it.
+            permalink = 'https://digitaltibet.ctl.columbia.edu/' + \
+                        'object/{}/'.format(filename.replace('_', '-'))
+            source.url = permalink
+            source.save()
+
+
+def overwrite_with_primary_sources(Source, sources):
+    for source in sources:
```

Comment:
Delving into this issue on my local dev env -- the sources passed through are of type `xyztile`. The end of the source here `x${x}` is a format string that is replaced by a given tile in `sherdjs/src/image/views/openlayers.js`. Replacing these with the primary source url will break the tiling mechanism. I think we'll need to make sure all the tiles are still available in the hugo-fied Digital Tibet, and then just replace the s3 bucket url with the https://digitaltibet.ctl.columbia.edu/ per the `fix_permalinks` function.